src/Nethermind/Nethermind.Merge.Plugin/Synchronization/MergeBlockDownloader.cs

```diff
@@ -275,6 +283,35 @@ bool HasMoreToSync()
             return blocksSynced;
         }
 
+        private Dictionary<Keccak, Block> GetExistingBlocks(bool downloadReceipts, IReadOnlyList<BlockHeader> headers)
+        {
+            Dictionary<Keccak, Block> knownBlocks = new();
+
+            if (downloadReceipts)
+            {
+                // If download receipt is turned on, we don't set known blocks, because it will skip downloading
+                // receipts. Maybe later we'll have knownReceipts also.
+                return knownBlocks;
+            }
+
+            for (int i = 0; i < headers.Count; i++)
+            {
+                if (_blockCacheService.BlockCache.TryGetValue(headers[i].Hash!, out Block? block))
```

Comment:
Do we pass hive tests without this check?

Reply:
Dont think so,